net/instaweb/rewriter/csp_directive.gperf

```diff
@@ -0,0 +1,44 @@
+%{
+// csp_directive.gp.cc is automatically generated from csp_directive.gperf
+// Author: morlovich@google.com
+
+#include "base/basictypes.h"
+#include "net/instaweb/rewriter/public/csp_directive.h"
+#include "pagespeed/kernel/base/string_util.h"
+
+namespace net_instaweb {
+%}
+%compare-strncmp
+%define class-name CspDirectiveMapper
+%define lookup-function-name Lookup
+%define word-array-name CspDirectiveNameTable
+%global-table
```

Comment:
Possibly not, I just wanted flexibility before I sorted out exactly what portion of the spec we care about. 
(Actually there are other specs that may be relevant, this thing is modular...)